flytekit/annotated/python_function_task.py

```diff
@@ -167,9 +177,17 @@ def __init__(
             **kwargs,
         )
         self._task_function = task_function
+        self._execution_mode = execution_mode
+
+    @property
+    def execution_mode(self) -> ExecutionBehavior:
+        return self._execution_mode
 
     def execute(self, **kwargs) -> Any:
-        return self._task_function(**kwargs)
+        if self.execution_mode == self.ExecutionBehavior.DEFAULT:
+            return self._task_function(**kwargs)
+        elif self.execution_mode == self.ExecutionBehavior.DYNAMIC:
+            return self.dynamic_execute(self._task_function, **kwargs)
```

Comment:
ohh this is interesting. This will break for cases in which execute is overriden right? Which happens quite a few times? Are we sure this is the best place to do it?

Reply:
so the contract is that `execute` is overridable. but dispatch_execute should not be overriden

Reply:
I guess now that I have added pre-execute and post-execute, we can document that `execute` should not be overriden in most cases unless absolutely required and in that case, unless you replicate the dynamic execute call, ability to dynamically execute will be lost! I am ok with that

Reply:
> so the contract is that execute is overridable. but dispatch_execute should not be overriden

yes precisely. added documentation that `execute` can be overwritten if dynamic execute is handled too